File: epi_judge/swap_bits.cc
#include "test_framework/generic_test.h"
long long SwapBits(long long x, int i, int j) {
    if (((x >> i) & 1) ^ ((x >> j) & 1)) {
        return x ^ (((long long)1<<i)|((long long)1<<j));
    }
    return x;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"x", "i", "j"};
  return GenericTestMain(args, "swap_bits.tsv", &SwapBits, DefaultComparator{},
                         param_names);
}

File: epi_judge/swap_bits.py
def swap_bits(x, i, j):
    bit_i = (x >> i) & 1
    bit_j = (x >> j) & 1

    swap_bit = bit_i ^ bit_j
    if not swap_bit:
        return x

    mask = (swap_bit << i) | (swap_bit << j)
    return x ^ mask


from sys import exit

from test_framework import generic_test, test_utils

if __name__ == '__main__':
    exit(generic_test.generic_test_main('swap_bits.tsv', swap_bits))
